pkg/specgen/generate/pod_create.go

```diff
@@ -176,6 +176,7 @@ func createPodOptions(p *specgen.PodSpecGenerator, rt *libpod.Runtime, infraSpec
 			options = append(options, libpod.WithPodUser())
 		}
 	}
+	options = append(options, libpod.WithPodCgroups())
```

Comment:
`WithPodCgroup` or `Cgroups`?

Reply:
also, should this always be done?? what if we do not want to share?

Reply:
Agreed we should only do this is the pod shares cgroups.

Reply:
`WithPodCgroups` used to be always called in v3.3:
  https://github.com/containers/podman/blob/v3.3/pkg/specgen/generate/pod_create.go#L142

As far as I investigated, it seems that this was unintentionally removed:
- https://github.com/containers/podman/pull/11102/files#diff-edb3bda86436d2a7bc55b34fa48c71156dc5e02d5f452012b1750efb15b3158fR160
- https://github.com/containers/podman/pull/12271/files#diff-edb3bda86436d2a7bc55b34fa48c71156dc5e02d5f452012b1750efb15b3158fL261

This is why I just restored this call.

I would consider introducing a new flag to specify whether cgroup is shared in a pod or not though it might be better to work in another PR.

Reply:
IT should be handled via the --shares command, if the user wants to eliminate it, then using the shares command should handle it.

Reply:
I think it should always be called by default as well @hshiina  but the issue here is that `--share=none` here will not take affect since line is executed always, right?

Reply:
I should have clarified which is expected to be shared by `pod create --share cgroup`, cgroup or cgroup namespace, at first. Though this PR assumes cgroup namespace is shared based on the help message, I would reconsider.